x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -498,6 +509,7 @@ private static List<String> replaceWildcardsWithAuthorizedAliases(String[] alias
         }
 
         for (String aliasExpression : aliases) {
+            IndexNameExpressionResolver.assertExpressionHasDefaultOrDataSelector(aliasExpression);
```

Comment:
AFAIK the data streams support aliases, for example: we can have `my-alias` -> `my-data-stream` -> `backing indices`. Is accessing failure store via data stream alias not supported? Meaning,  can users search against `my-alias::failures`?

Reply:
Querying via alias is indeed supported -- however, this method is only called for instances of `AliasesRequest` which currently are `AliasActions` and `GetAliasesRequest` neither of which support failure selectors.